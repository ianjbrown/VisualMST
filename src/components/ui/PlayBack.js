import React, { useState } from "react";
import classes from "./PlayBack.module.css";
import TooltipIcon from "./TooltipIcon";

import Slider from "@mui/material/Slider";
import { Col } from "react-bootstrap";

function PlayBack(props) {
  const [backgroundOn, setBackgroundOn] = useState(true);

  const timeoutHandler = (event, newValue) => {
    props.onTimeoutChange(newValue);
  };

  function firstHandler() {
    props.onFirst();
  }

  function backHandler() {
    props.onBack();
  }

  function pauseHandler() {
    props.onPause();
  }

  function forwardHandler() {
    props.onForward();
  }

  function lastHandler() {
    props.onLast();
  }

  function toggleBackgroundGraph() {
    props.onToggleBackgroundGraph(!backgroundOn);
    setBackgroundOn(!backgroundOn);
  }

  return (
    <>
      <Col data-testid="slider-col" xs={3} className={classes.bar}>
        <div data-testid="slow" className={classes.barText}>Slow</div>
        <Slider
          data-testid="slider"
          className={classes.slider}
          aria-label="Speed"
          defaultValue={1000}
          valueLabelDisplay="auto"
          valueLabelFormat={(value) => <div>Speed: {value / 200}</div>}
          step={100}
          marks
          min={100}
          max={2000}
          onChangeCommitted={timeoutHandler}
        />
        <div data-testid="fast" className={classes.barText}>Fast</div>
      </Col>
      <Col data-testid="playback-col" xs={6} className={classes.bar}>
        <TooltipIcon data-testid="first" id="first" firstHandler={firstHandler} />
        <TooltipIcon data-testid="back" id="back" backHandler={backHandler} />
        <TooltipIcon
          data-testid="play"
          id="play"
          pauseHandler={pauseHandler}
          paused={props.paused}
        />
        <TooltipIcon data-testid="forward" id="forward" forwardHandler={forwardHandler} />
        <TooltipIcon data-testid="last" id="last" lastHandler={lastHandler} />
      </Col>
      <Col data-testid="toggle-col" xs={3} className={classes.bar}>
        <TooltipIcon
          data-testid="hide"
          id="hide"
          toggleHandler={toggleBackgroundGraph}
          backgroundOn={backgroundOn}
        />
        <div data-testid="hide-text" className={classes.barText}>Hide Original Graph</div>
      </Col>
    </>
  );
}

export default PlayBack;
