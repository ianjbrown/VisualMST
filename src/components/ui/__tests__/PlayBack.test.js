import { render, screen, cleanup } from "@testing-library/react";
import PlayBack from "../PlayBack";
import "@testing-library/jest-dom/";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("PlayBack component matches snapshot", () => {
  const playBack = renderer.create(<PlayBack />).toJSON();
  expect(playBack).toMatchSnapshot();
});

test("Slider Col loads and has class bar", () => {
  render(<PlayBack />);
  const sliderCol = screen.getByTestId("slider-col");
  expect(sliderCol).toBeInTheDocument();
  expect(sliderCol).toHaveClass("bar");
  // expect(sliderCol).toContainElement(screen.getByTestId("slow"));
  // expect(sliderCol).toContainElement(screen.getByTestId("slider"));
  // expect(sliderCol).toContainElement(screen.getByTestId("fast"));
});

// test("Slow and Fast load and have class barText", () => {
//   render(<PlayBack />);
//   const slow = screen.getByTestId("slow");
//   const fast = screen.getByTestId("fast");
//   expect(slow).toBeInTheDocument();
//   expect(fast).toBeInTheDocument();
//   expect(slow).toHaveClass("barText");
//   expect(fast).toHaveClass("barText");
// });

// test("Slider loads and has class slider", () => {
//   render(<PlayBack />);
//   const slider = screen.getByTestId("slider");
//   expect(slider).toBeInTheDocument();
//   expect(slider).toHaveClass("slider");
// });

test("Playback Col loads and has class bar", () => {
  // const onPause = jest.fn();
  // const onTimeoutChange = jest.fn();
  // const onFirst = jest.fn();
  // const onBack = jest.fn();
  // const onForward = jest.fn();
  // const onLast = jest.fn();
  // const onToggleBackgroundGraph = jest.fn();
  render(
    <PlayBack
    // paused={true}
    // onPause={onPause}
    // onTimeoutChange={onTimeoutChange}
    // onFirst={onFirst}
    // onBack={onBack}
    // onForward={onForward}
    // onLast={onLast}
    // onToggleBackgroundGraph={onToggleBackgroundGraph}
    />
  );
  const playBackCol = screen.getByTestId("playback-col");
  expect(playBackCol).toBeInTheDocument();
  expect(playBackCol).toHaveClass("bar");
  // expect(playBackCol).toContainElement(screen.queryByTestId("first"));
  // expect(playBackCol).toContainElement(screen.getByTestId("back"));
  // expect(playBackCol).toContainElement(screen.getByTestId("play"));
  // expect(playBackCol).toContainElement(screen.getByTestId("forward"));
  // expect(playBackCol).toContainElement(screen.getByTestId("last"));
});

test("Toggle Col loads and has class bar", () => {
  render(<PlayBack />);
  const toggleCol = screen.getByTestId("toggle-col");
  expect(toggleCol).toBeInTheDocument();
  expect(toggleCol).toHaveClass("bar");
  // expect(toggleCol).toContainElement(screen.getByTestId("hide"));
  // expect(toggleCol).toContainElement(screen.getByTestId("hide-text"));
});

// test("Hide original graph loads and has class barText", () => {
//   render(<PlayBack />);
//   const barText = screen.getByTestId("hide-text");
//   expect(barText).toBeInTheDocument();
//   expect(barText).toHaveClass("barText");
// });
