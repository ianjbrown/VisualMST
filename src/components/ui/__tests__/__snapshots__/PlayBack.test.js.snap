// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayBack component matches snapshot 1`] = `
Array [
  <div
    className="bar col-3"
    data-testid="slider-col"
  >
    <div
      className="barText"
      data-testid="slow"
    >
      Slow
    </div>
    <span
      className="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root slider css-187mznn-MuiSlider-root"
      data-testid="slider"
      onMouseDown={[Function]}
    >
      <span
        className="MuiSlider-rail css-14pt78w-MuiSlider-rail"
      />
      <span
        className="MuiSlider-track css-1gv0vcd-MuiSlider-track"
        style={
          Object {
            "left": "0%",
            "width": "47.36842105263158%",
          }
        }
      />
      <span
        className="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index={0}
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <span
        className="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index={1}
        style={
          Object {
            "left": "5.2631578947368425%",
          }
        }
      />
      <span
        className="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index={2}
        style={
          Object {
            "left": "10.526315789473685%",
          }
        }
      />
      <span
        className="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index={3}
        style={
          Object {
            "left": "15.789473684210526%",
          }
        }
      />
      <span
        className="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index={4}
        style={
          Object {
            "left": "21.05263157894737%",
          }
        }
      />
      <span
        className="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index={5}
        style={
          Object {
            "left": "26.31578947368421%",
          }
        }
      />
      <span
        className="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index={6}
        style={
          Object {
            "left": "31.57894736842105%",
          }
        }
      />
      <span
        className="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index={7}
        style={
          Object {
            "left": "36.8421052631579%",
          }
        }
      />
      <span
        className="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index={8}
        style={
          Object {
            "left": "42.10526315789474%",
          }
        }
      />
      <span
        className="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index={9}
        style={
          Object {
            "left": "47.36842105263158%",
          }
        }
      />
      <span
        className="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index={10}
        style={
          Object {
            "left": "52.63157894736842%",
          }
        }
      />
      <span
        className="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index={11}
        style={
          Object {
            "left": "57.89473684210526%",
          }
        }
      />
      <span
        className="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index={12}
        style={
          Object {
            "left": "63.1578947368421%",
          }
        }
      />
      <span
        className="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index={13}
        style={
          Object {
            "left": "68.42105263157895%",
          }
        }
      />
      <span
        className="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index={14}
        style={
          Object {
            "left": "73.6842105263158%",
          }
        }
      />
      <span
        className="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index={15}
        style={
          Object {
            "left": "78.94736842105263%",
          }
        }
      />
      <span
        className="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index={16}
        style={
          Object {
            "left": "84.21052631578948%",
          }
        }
      />
      <span
        className="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index={17}
        style={
          Object {
            "left": "89.47368421052632%",
          }
        }
      />
      <span
        className="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index={18}
        style={
          Object {
            "left": "94.73684210526316%",
          }
        }
      />
      <span
        className="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index={19}
        style={
          Object {
            "left": "100%",
          }
        }
      />
      <span
        className="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
        data-index={0}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "left": "47.36842105263158%",
            "pointerEvents": undefined,
          }
        }
      >
        <input
          aria-label="Speed"
          aria-orientation="horizontal"
          aria-valuemax={2000}
          aria-valuemin={100}
          aria-valuenow={1000}
          data-index={0}
          disabled={false}
          max={2000}
          min={100}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          step={100}
          style={
            Object {
              "border": 0,
              "clip": "rect(0 0 0 0)",
              "direction": "ltr",
              "height": "100%",
              "margin": -1,
              "overflow": "hidden",
              "padding": 0,
              "position": "absolute",
              "whiteSpace": "nowrap",
              "width": "100%",
            }
          }
          type="range"
          value={1000}
        />
        <span
          aria-hidden={true}
          className="MuiSlider-valueLabel css-1kz0hui-MuiSlider-valueLabel"
        >
          <span
            className="MuiSlider-valueLabelCircle"
          >
            <span
              className="MuiSlider-valueLabelLabel"
            >
              <div>
                Speed: 
                5
              </div>
            </span>
          </span>
        </span>
      </span>
    </span>
    <div
      className="barText"
      data-testid="fast"
    >
      Fast
    </div>
  </div>,
  <div
    className="bar col-6"
    data-testid="playback-col"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1tvxjov-MuiSvgIcon-root"
      data-testid="FirstPageIcon"
      focusable="false"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1tvxjov-MuiSvgIcon-root"
      data-testid="ArrowBackIosNewOutlinedIcon"
      focusable="false"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M17.77 3.77 16 2 6 12l10 10 1.77-1.77L9.54 12z"
      />
    </svg>
    <div
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1tvxjov-MuiSvgIcon-root"
        data-testid="PauseOutlinedIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"
        />
      </svg>
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1tvxjov-MuiSvgIcon-root"
      data-testid="ArrowForwardIosOutlinedIcon"
      focusable="false"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1tvxjov-MuiSvgIcon-root"
      data-testid="LastPageIcon"
      focusable="false"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
      />
    </svg>
  </div>,
  <div
    className="bar col-3"
    data-testid="toggle-col"
  >
    <div
      className="barToggle"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1tvxjov-MuiSvgIcon-root"
        data-testid="ToggleOffOutlinedIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
        />
      </svg>
    </div>
    <div
      className="barText"
      data-testid="hide-text"
    >
      Hide Original Graph
    </div>
  </div>,
]
`;
